equinix/provider.go

```diff
@@ -44,18 +45,25 @@ func Provider() *schema.Provider {
 				Description:  "The Equinix API base URL to point out desired environment. Defaults to https://api.equinix.com",
 			},
 			"client_id": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				DefaultFunc:  schema.EnvDefaultFunc(clientIDEnvVar, nil),
-				ValidateFunc: validation.StringIsNotEmpty,
-				Description:  "API Consumer Key available under My Apps section in developer portal",
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc(clientIDEnvVar, nil),
+				// ValidateFunc: validation.StringIsNotEmpty,
+				Description: "API Consumer Key available under My Apps section in developer portal",
 			},
 			"client_secret": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				DefaultFunc:  schema.EnvDefaultFunc(clientSecretEnvVar, nil),
-				ValidateFunc: validation.StringIsNotEmpty,
-				Description:  "API Consumer secret available under My Apps section in developer portal",
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc(clientSecretEnvVar, nil),
+				// ValidateFunc: validation.StringIsNotEmpty,
+				Description: "API Consumer secret available under My Apps section in developer portal",
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc(clientTokenEnvVar, nil),
+				// ValidateFunc: validation.StringIsNotEmpty,
+				Description: "API token from the developer sandbox",
```

Comment:
It shall be valid also for production environment, previously retrieving a token with the endpoint https://api.equinix.com/oauth2/v1/token . I wonder if we should update /docs to explain this attribute and its possibilities